torchaudio/functional/functional.py

```diff
@@ -1678,6 +1678,92 @@ def pitch_shift(
     return waveform_shift
 
 
+def _pitch_shift_preprocess(
+    waveform: Tensor,
+    n_steps: int,
+    bins_per_octave: int = 12,
+    n_fft: int = 512,
+    win_length: Optional[int] = None,
+    hop_length: Optional[int] = None,
+    window: Optional[Tensor] = None,
+) -> Tensor:
+    """
+    transforms PitchShift helper function to preprocess before resampling step.
+
+    .. devices:: CPU CUDA
```

Comment:
no need to include devices and properties -- these are displayed for users to know what is supported, but these functions are not user facing